### PR TITLE
Add nowrap-expanded prop to VDataTable

### PR DESCRIPTION
**Pull request by dimailn**
## Description
Adding new prop  no-wrapped-expand-slots that setted to true prevents wrap slot contents to tr/td elements.

## Motivation and Context
Sometimes we need render current table rows in expand slot, without breaking view(because now tr's in expand slot renders as another table, independent of main table.

Unfortunately, I have no idea how to solve problem with saving transition, I'd appreciate some help with this.

## How Has This Been Tested?
https://codepen.io/anon/pen/RMyBzm

## Markup:
```vue
<v-data-table
  :headers="headers"
  :items="items"
  hide-actions
  item-key="name"
  no-wrapped-expand-slots
>
  <template slot="items" slot-scope="props">
    <tr @click="props.expanded = !props.expanded">
      <td>{{ props.item.name }}</td>
      <td class="text-xs-right">{{ props.item.calories }}</td>
      <td class="text-xs-right">{{ props.item.fat }}</td>
      <td class="text-xs-right">{{ props.item.carbs }}</td>
      <td class="text-xs-right">{{ props.item.protein }}</td>
      <td class="text-xs-right">{{ props.item.iron }}</td>
    </tr>
  </template>
  <template slot="expand" slot-scope="props">
    <tr v-for="ingredient in props.item.ingredients" :key="ingredient.id">
      <td>{{ ingredient.name }}</td>
      <td class="text-xs-right">{{ ingredient.calories }}</td>
      <td class="text-xs-right">{{ ingredient.fat }}</td>
      <td class="text-xs-right">{{ ingredient.carbs }}</td>
      <td class="text-xs-right">{{ ingredient.protein }}</td>
      <td class="text-xs-right">{{ ingredient.iron }}</td>
    </tr>
  </template>
</v-data-table>
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

**Comment by johnleider** [comment]
You need to provide an entire playground that works. The above needs data to properly show your example.

**Comment by dimailn** [comment]
https://codepen.io/anon/pen/RMyBzm

File: src/components/VDataTable/mixins/body.js [hunk 1]
```diff
@@ -28,7 +33,7 @@ export default {
         on: ExpandTransitionGenerator('datatable__expand-col--expanded')
       }, children)
 
-      return this.genTR([transition], { class: 'datatable__expand-row' })
+      return this.nowrapExpanded ? children : this.genTR([transition], { class: 'datatable__expand-row' })
```
Comment (nekosaur) [inline src/components/VDataTable/mixins/body.js]: Move the return if nowrapExpanded to above line 27 so we're not creating an unused transition-group every time.

**Comment by dimailn** [comment]
ready

**Comment by dimailn** [comment]
@johnleider @nekosaur

**Comment by dimailn** [comment]
=)

**Comment by johnleider** [comment]
I know a lot of work has gone into this. With the introduction of #3833 though, I'm freezing all feature development on this component.

**Pull request closed without merging**